scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -196,115 +197,128 @@ public int schedule() {
-                    //Something goes wrong with the active object creation (createLauncher)
-                    logger.warn("An exception occured while creating the task launcher.", e1);
-                    //so try to get back every remaining nodes to the resource manager
-                    try {
-                        releaseNodes(currentJob, nodeSet);
-                        freeResources.addAll(nodeSet.getAllNodesUrls());
-                    } catch (Exception e2) {
-                        logger.info("Unable to get back the nodeSet to the RM", e2);
-                    }
-                    if (--activeObjectCreationRetryTimeNumber == 0) {
-                        return numberOfTaskStarted;
-                    }
-                } catch (Exception e1) {
-                    //if we are here, it is that something append while launching the current task.
-                    logger.warn("An exception occured while starting task.", e1);
-                    //so try to get back every remaining nodes to the resource manager
-                    try {
-                        releaseNodes(currentJob, nodeSet);
-                        freeResources.addAll(nodeSet.getAllNodesUrls());
-                    } catch (Exception e2) {
-                        logger.info("Unable to get back the nodeSet to the RM", e2);
+                    } catch (ActiveObjectCreationException e1) {
+                        //Something goes wrong with the active object creation (createLauncher)
+                        logger.warn("An exception occured while creating the task launcher.", e1);
+                        //so try to get back every remaining nodes to the resource manager
+                        try {
+                            releaseNodes(currentJob, nodeSet);
+                            freeResources.addAll(nodeSet.getAllNodesUrls());
+                        } catch (Exception e2) {
+                            logger.info("Unable to get back the nodeSet to the RM", e2);
+                        }
+                        if (--activeObjectCreationRetryTimeNumber == 0) {
+                            return numberOfTaskStarted;
+                        }
+                    } catch (Exception e1) {
+                        //if we are here, it is that something append while launching the current task.
+                        logger.warn("An exception occured while starting task.", e1);
+                        //so try to get back every remaining nodes to the resource manager
+                        try {
+                            releaseNodes(currentJob, nodeSet);
+                            freeResources.addAll(nodeSet.getAllNodesUrls());
+                        } catch (Exception e2) {
+                            logger.info("Unable to get back the nodeSet to the RM", e2);
+                        }
                     }
                 }
+                subTasksListFromInclusive += subTasksListToExclusive;
```

Comment:
it is not clear to me why this values are chosen. Maybe if the logic is exported in a dedicated class, it will be more understandable